homeassistant/components/sensor/envirophat.py

```diff
@@ -53,6 +54,14 @@
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the Sense HAT sensor platform."""
+    # Detect platform by checking for the onboard sensors showing up on i2c
+    if (subprocess.getoutput("i2cdetect -q -y 1 | grep -o 1d") != '1d' and
```

Comment:
Instead of this crazy hack, can't you just use the `data` object below to update and get an error when it fails?